### ISSUE TITLE
Object navigation gets stuck Outlook 2013

### ISSUE THREAD

**Issue by feerrenrut** [nvaccess/nvda#7782]
### Steps to reproduce:
1. With `simple review mode` enabled (NVDA menu, preferences, review cursor)
1. Open outlook 2013
1. Use `NVDA+numpad up` to get to the "Outlook today - Outlook window"
1. Press `NVDA+numpad down`  and the following " System  menu bar  Alt+Space"
1. Then press `NVDA+numpad right` and get the following: 
   1. "Minimise  button  Moves the window out of the way"
   1. "Restore  button  Puts a minimised or maximised window back to normal"
   1. "Close  button  Closes the window"
   1. "Ribbon  tool bar"
   1. "Status Bar  tool bar"
   1. "Ribbon  tool bar"

### Expected behavior:
There are other elements that you should be able to navigate to, including:
- Favourites tree
- Navigation bar ("Mail", "Calendar", "People", "Tasks")
- Outlook Today pain

### Actual behavior:
While continuing to press `NVDA+numpad right` the "Status Bar tool bar" and "Ribbon tool bar" are repeated. 
When trying to go back with `NVDA+numpad left` the same items are repeated.

### System configuration:
NVDA version: next-14617,ef52ddc4

NVDA Installed or portable: Installed

Name and version of other software in use when reproducing the issue: Outlook, MS office 2013

### Other information
With simple object navigation off, I can navigate to all of the expected elements.

**Comment by Adriani90** [comment]
I encounter the same issue where Object nav repeats the objects on that level in outlook 2016 on Windows 10 as well.

**Comment by didiercolle** [comment]
As indicated in issue #7759 , I have similar problems in Thunderbird and Adobe Acrobat Reader DC.

Configuration:
NVDA: installed version 2017.4, simple object navigation on.
OS: Windows 10
Adobe Acrobat Reader DC: version 18.0, Continuous Release | Version 2018.011.20036
Thunderbird: 52.6.0 (32-bit)

Reproducing one of the problems in Adobe Acrobat Reader DC:
1. Make sure Adobe Acrobat Reader DC is closed
2. Double click a pdf file
3. navigate 9 times left (nvda+shift+left arrow in laptop layout)
4. "learn more button" is spoken.
5. keep navigating left: you keep jumping/cycling between this "learn more button" and "store and share files in the document cloud button"

The Thunderbird problem:
1. Open Thunderbird with following screen layout: top half the table with all emails in a mail folder, bottom half displaying the content of a selected email, left the hierarchy tree of all mail folders
2. click/select on a mail in the table with all mails in the selected folder
3. navigate 4 times up, such that I believe "<selected folder> - <account name or email address> window" is spoken, signaling you have reached the top window for Thunderbird.
4. navigate down one level
5. navigate 4 times right (the same is spoken as in point 3, except busy instead of window is spoken at the end)
6. try to navigate left: "no previous" is spoken.
7. navigate up once and repeat from step 4, showing that circling around is possible forever, while not in the other direction.

**Comment by didiercolle** [comment]
I had a look at the navigation code: that looks ok to me.

It seems that the object hierarchy is broken.
I wrote some helper functions to check the consistency of the object hierarchy.
You can import it as follows:
1. open python console: press nvda+ctrl+z
2. type: from sys import path
3. type and  replace between '' the path to where you saved the helper.py file (simply remove the .txt extension): path.insert (0, r'c:\cygwin\home\dcolle\src\tmp')
4. type: from helper import *
5. type: checkAncestors(nav)
What the latter function call does is basically the following: create the path from the current navigation object (nav) towards the top window of the app, and for each of these ancestors checks the different relationships from its children: the first child is supposed not to have a previous ref, the last one is supposed not to have a next ref, child with index i is supposed to have a previous ref pointing to child with index i-1 and a next ref pointing to child with index i+1, and the parent ref should point to the parent node under investigation. In case problems in this respect are found, an error message is printed together with some additional info on the problematic ref.

Attached are the outputs for:
1. Thunderbird problem: execution after step 2 in the problem description in previous comment.
2. Adobe Acrobat Reader DC problem: execution after step 3 in the problem description in previous comment.

[helper.py.txt](https://github.com/nvaccess/nvda/files/1733961/helper.py.txt)
[outputAdobe.txt](https://github.com/nvaccess/nvda/files/1733962/outputAdobe.txt)
[outputThunderbird.txt](https://github.com/nvaccess/nvda/files/1733963/outputThunderbird.txt)

**Comment by Adriani90** [comment]
I think this issue has to be edited since we can reproduce this issue in several applications. I have just discovered that this issue arises also in MS PowerPoint and other applications. I don't know if this is up to NVDA. I rather think that the developer must build in a consistent object jerarchy. But I think in case of MS Office we could start conversations with Microsoft to improve object jerarchy since many users would Profit out of it. That is for example in MS PowerPoint we could overview a slide much faster that navigating individual objects by pressing tab or shift+tab. I think it is very hard to implement a NVDA specific object jerarchy.

Comments are appreciated.

**Comment by Adriani90** [comment]
We have to find out in which regions of the window NVDA gets stuck. According to my experience it is often either near to Status bar or to ribbons.

**Comment by Adriani90** [comment]
In my view the accessibility Impact would be much higher if developers developed a Logical and consistent object hierarchy. I don't know if there are any accessibility Standards related to this. However, I have noticed that NVDA does not get stuck in Outlook 2010. I think it might be related to UIA.

**Comment by josephsl** [comment]
Hi,

Eight years later...

Outlook (or for that matter, Office) 2013 is out of scope. So is this issue reproducible with newer Outlook releases such as Outlook 2024?

Thanks.
